src/features/defi/providers/osmosis/components/OsmosisManager/Deposit/components/Deposit.tsx

```diff
@@ -0,0 +1,375 @@
+    if (!(contextDispatch && userAddress && assetReference && accountId && opportunity)) return
+    try {
+      const chainAdapters = getChainAdapterManager()
+      const adapter = chainAdapters.get(
+        chainId,
+      ) as unknown as CosmosSdkBaseAdapter<CosmosSdkChainId>
+      const fastFeeCryptoBaseUnit = (
+        await adapter.getFeeData({
+          sendMax: false,
+        })
+      ).fast.txFee
+
+      const fastFeeCryptoPrecision = bnOrZero(
+        bn(fastFeeCryptoBaseUnit).div(bn(10).pow(asset?.precision ?? 0)),
+      )
+      return bnOrZero(fastFeeCryptoPrecision).toString()
+    } catch (error) {
+      moduleLogger.error(
+        { fn: 'getDepositFeeEstimate', error },
+        'Error getting deposit fee estimate',
+      )
+      toast({
+        position: 'top-right',
+        description: translate('common.somethingWentWrongBody'),
+        title: translate('common.somethingWentWrong'),
+        status: 'error',
+      })
+    } finally {
+      contextDispatch({ type: OsmosisDepositActionType.SET_LOADING, payload: false })
+    }
+  }, [
+    contextDispatch,
+    userAddress,
+    assetReference,
+    accountId,
+    opportunity,
+    asset,
+    chainId,
+    toast,
+    translate,
+  ])
+
+  /** Returns an array containing the allocation fraction and shareOutAmount,
+   * calculated from the form input parameters */
+  /** Share out amount is the exact number of tokens expected to be received. See documentation below:
+   * https://docs.osmosis.zone/osmosis-core/modules/gamm/#join-pool
+   * https://github.com/osmosis-labs/osmosis-frontend/blob/d3badcee48f54e61e6a6117c294c0725822a4100/packages/stores/src/ui-config/manage-liquidity/add-liquidity.ts#L371
+   * NOTE: This calculation may need to be modified to support pools with unequal weights at a later date. The Osmosis frontend does not currently do this.
+   */
+  const calculateAllocations = useCallback(
+    (
+      inputAsset: Asset,
+      inputAssetAmount: string,
+    ): { allocationFraction: string; shareOutAmount: string } => {
+      const ret = { allocationFraction: '0', shareOutAmount: '0' }
+      if (!state || !state.poolData) {
+        return ret
+      }
+      const poolAssets = state?.poolData?.pool_assets
+
+      let { assetReference: inputAssetReference } = fromAssetId(inputAsset.assetId)
+      if (inputAssetReference === ASSET_REFERENCE.Osmosis) inputAssetReference = 'uosmo'
+
+      if (!poolAssets) return ret
+      const poolAssetIndex = poolAssets.findIndex(asset =>
+        asset.token.denom.toLowerCase().includes(inputAssetReference.toLowerCase()),
+      )
+      if (poolAssetIndex < 0) return ret
+
+      const inputAssetAmountBaseUnits = bnOrZero(inputAssetAmount)
+        .pow(10, bnOrZero(inputAsset.precision))
+        .toString()
+
+      const poolTotalSharesBaseUnits = state?.poolData?.total_shares?.amount
+      if (!poolTotalSharesBaseUnits) {
```

Comment:
Check at the very beginning of the callback body after l.144
Also at this point `state.poolData` should be defined so opt chain can be avoided